galsim/interpolatedimage.py

```diff
@@ -550,6 +546,182 @@ def __setstate__(self, d):
+    their real-space profiles (e.g., in InterpolatedImage), though no extensive testing has been
+    performed for objects specified by samples of their Fourier-space profiles (e.g., this
+    class).
+
+    Initialization
+    --------------
+
+        >>> interpolated_kimage = galsim.InterpolatedKImage(real_kimage, imag_kimage,
+                k_interpolant=None, stepk=0., gsparams=None)
+
+    Initializes `interpolated_kimage` as an InterpolatedKImage instance.
+
+    @param real_kimage      The Image corresponding to the real part of the Fourier-space samples.
+    @param imag_kimage      The Image corresponding to the imaginary part of the Fourier-space
+                            samples.
+    @param k_interpolant    Either an Interpolant instance or a string indicating which k-space
+                            interpolant should be used.  Options are 'nearest', 'sinc', 'linear',
+                            'cubic', 'quintic', or 'lanczosN' where N should be the integer order
+                            to use.  [default: galsim.Quintic()]
+    @param stepk            By default, the stepk value (the sampling frequency in Fourier-space)
+                            of the underlying SBProfile is set by the `scale` attribute of the
+                            supplied images.  This keyword allows the user to specify a coarser
+                            sampling in Fourier-space, which may increase efficiency at the expense
+                            of decreasing the separation between neighboring copies of the
+                            DFT-rendered real-space profile.  (See the GSParams docstring for the
+                            parameter `folding_threshold` for more information).
+                            [default: real_kimage.scale]
+    @param gsparams         An optional GSParams argument.  See the docstring for GSParams for
+                            details. [default: None]
+
+    Methods
+    -------
+
+    There are no additional methods for InterpolatedKImage beyond the usual GSObject methods.
+    """
+    _req_params = { 'real_kimage' : str,
+                    'imag_kimage' : str }
+    _opt_params = {
+        'k_interpolant' : str,
+        'stepk': float
+    }
+    _single_params = []
+    _takes_rng = False
+    _takes_logger = False
+
+    def __init__(self, real_kimage, imag_kimage, k_interpolant=None, stepk=None,
+                 gsparams=None):
+
+        # make sure real_kimage, imag_kimage are really `Image`s, are floats, and are congruent.
+        if not isinstance(real_kimage, galsim.Image) or not isinstance(imag_kimage, galsim.Image):
+            raise ValueError("Supplied kimage is not an Image instance")
+        if ((real_kimage.dtype != np.float32 and real_kimage.dtype != np.float64)
+            or (imag_kimage.dtype != np.float32 and imag_kimage.dtype != np.float64)):
+            raise ValueError("Supplied image is not an image of floats or doubles!")
```

Comment:
"double" is a C++ term.  Just float is fine here for the error text, even though we also allow np.float64.  Or you can write out `float32 or float64`.  These are the terms we use in the Image class doc string.  Or perhaps "floating point Image" would be similarly descriptive.